mirgecom/euler.py

```diff
@@ -76,31 +75,40 @@
 class ConservedVars:  # FIXME: Name?
     r"""Resolve the canonical conserved quantities.
 
-    Get the canonical conserved quantities (mass, energy, momentum)
-    per unit volume = $(\rho,\rho{E},\rho\vec{V})$ from an agglomerated
-    object array.
+    Get the canonical conserved quantities (mass, energy, momentum,
+    and mass fractions) per unit volume = $(\rho,\rho{E},\rho\vec{V},
+    \rho{Y_s})$ from an agglomerated object array.
 
     .. attribute:: dim
 
+        Integer indicating spatial dimension of the state
+
     .. attribute:: mass
 
-        Mass per unit volume
+        :class:`~meshmode.dof_array.DOFArray` for the scalar mass per unit volume
 
     .. attribute:: energy
 
-        Energy per unit volume
+        :class:`~meshmode.dof_array.DOFArray` for total energy per unit volume
 
     .. attribute:: momentum
 
-        Momentum vector per unit volume
+        Object array of :class:`~meshmode.dof_array.DOFArray` for momentum density
+        of shape ``(ndim,)``
+
+    .. attribute:: mass_fractions
+
+        Object array of :class:`~meshmode.dof_array.DOFArray` for species mass
+        fractions, $\rho~Y_\alpha$ shape=(nspecies,).
```

Comment:
You aren't the only one who picked up on this.   Technically, it should be just "scalars," as you mention, and should include any transported scalar (including non-mixture ones, e.g. like marker particles or ionic species for plasmas).  

The main issue I have with changing it here is that it means we'll need to "do something" to then interpret part of the "scalar" array as mixture species, and some not (at least in name).   Since our current use of this part of the Q array is as mixture species *only* - I sought to allow this slight misnomer until we have non-mixture scalars that would warrant such a distinction.

Reply:
My main complaint is just that it's confusing to name it `mass_fractions` when it's really `rho * mass_fractions`. `energy` has this problem too to an extent, but you can kind of hand-wave it away by interpreting it as "energy component of CV" (`rho*E`) instead of literally "energy" (`E`). It's harder (for me at least) to do that with `mass_fractions`. Naming it something like `scalar` would make it more symmetric, IMO:

mass [component of CV]: `rho`
momentum [component of CV]: `rho * V`
energy [component of CV]: `rho * E`
scalar [component of CV]: `rho * Y`

🤷

Reply:
Alright, I'm convinced and will change it.

Reply:
Changing here #191